File: Checkout/Javascript/get_order.js
const https = require('https');
const crypto = require('crypto');
const fs = require('fs').promises;

const merchantId = "CHECKOUT_MERCHANT_ID";
const secretWord = "CHECKOUT_SECRET_KEY";

class TestClass {
    createTimestamp() {
        return new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '');
    }

    createAuthorizationToken() {
        const timestamp = this.createTimestamp();
        const hash = crypto.createHash('sha512');
        hash.update("" + secretWord + timestamp);
        return Buffer.from(merchantId + ':' + hash.digest('hex')).toString('base64');
    }

    getHeaders(extraHeaders = {}) {
        const timestamp = this.createTimestamp();
        const authToken = this.createAuthorizationToken();
        let headers = {
            'Authorization': `Svea ${authToken}`,
            'Timestamp': timestamp,
            ...extraHeaders
        };
        return headers;
    }

    async makeGetRequest(url, headers) {
        return new Promise((resolve, reject) => {
            https.get(url, { headers }, (resp) => {
                let data = '';
                resp.on('data', (chunk) => {
                    data += chunk;
                });
                resp.on('end', () => {
                    resolve(data);
                });
            }).on("error", (err) => {
                reject("Error: " + err.message);
            });
        });
    }
}

async function main() {
    let orderId = "";
    try {
        const orderIdData = await fs.readFile('./created_order_id.txt', 'utf8');
        orderId = orderIdData.trim();
        //console.log(`Using OrderId: ${orderId}`);
    } catch (err) {
        console.error(`Failed to read OrderId from file: ${err.message}`);
        return;
    }

    console.log("Running GET request for Checkout (Javascript)");
    const testInstance = new TestClass();
    const myHeaders = testInstance.getHeaders({'Content-Type': 'application/json'});
    //const url = `https://paymentadminapistage.svea.com/api/v1/orders/${orderId}`;
    const url = `https://checkoutapistage.svea.com/api/orders/${orderId}`;

    testInstance.makeGetRequest(url, myHeaders)
        .then(response => {
            //console.log("Response: " + response + "\n");
            if (response) {
                console.log("Success!");
            } else {
                console.log("Failed...");
            }
        })
        .catch(error => console.log(error));
}

main();

File: Checkout/Python/get_order.py
from datetime import datetime
from datetime import timezone
from pandas import json_normalize
import hashlib
import base64
import requests
from pathlib import Path

merchant_id = "CHECKOUT_MERCHANT_ID"
secret_word = "CHECKOUT_SECRET_KEY"

class TestClass:

    def __init__(self):
        pass

    def get_request_headers(self, requests_body="", extra_headers=None):
        timestamp: str = datetime.now(timezone.utc).strftime("%Y-%m-%d %H:%M:%S")
        token = self.get_auth_token(timestamp, requests_body)
        #print(token)
        headers = {"Authorization": f"Svea {token}", "Timestamp": timestamp}
        if extra_headers:
            headers.update(extra_headers)
        return headers

    def get_auth_token(self, timestamp, request_body="") -> str:
        digest = request_body + secret_word + timestamp
        h = hashlib.sha512(digest.encode("utf-8")).hexdigest()
        auth = f"{merchant_id}:{h}"
        token = base64.b64encode(f"{auth}".encode('ascii')).decode('ascii')
        return token

print("Running GET request for Checkout (Python)")
testInstance = TestClass()
#my_headers = testInstance.get_request_headers(extra_headers={'Content-Type':'application/json'})
my_headers = testInstance.get_request_headers()
#print(my_headers)

order_id = ""
try:
    order_id_path = Path("./created_order_id.txt")
    with open(order_id_path, "r") as file:
        order_id = file.read().strip()
    #print(f"Using OrderId: {order_id}")
except FileNotFoundError:
    print("Error: OrderId file not found.")
    exit()
except Exception as e:
    print(f"Error reading OrderId: {e}")
    exit()

url = "https://checkoutapistage.svea.com/api/orders/" + order_id
#url = "https://paymentadminapistage.svea.com/api/v1/orders/" + order_id
payload = {}

#print("Fetching order:", order_id)
response = requests.request("GET", url, headers=my_headers, data=payload)
#response = requests.get(url, headers=my_headers, data=payload)
#print(response.text)
#response.raise_for_status()

if response.status_code == 200:
    print("Success!")
else:
    print("Failed...")

if response:
    dictr = response.json()
    result = json_normalize(dictr)
    #print(result)

